compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -8140,6 +8140,7 @@ private BType checkOptionalFieldAccessExpr(BLangFieldBasedAccess fieldAccessExpr
             fieldAccessExpr.originalType = fieldAccessExpr.leafNode || !nillableExprType ? actualType :
                     types.getTypeWithoutNil(actualType);
         } else if (types.isLax(effectiveType)) {
+            effectiveType = Types.getReferredType(varRefType);
```

Comment:
With this we disallow following kind of scenarios also
```ballerina
public function main() {
    map<map<json>> j = { a: { b: 1} };
    json _ = j?.a?.b;
}
```

Reply:
Few tests failed related to that scenario.

Reply:
```
if (effectiveType.tag == TypeTags.TYPEREFDESC) {
     effectiveType = Types.getReferredType(effectiveType);
}

```
This will solve the issue

Reply:
Fixed this. 4bad3ce58423eb609f5807788e4c83cf238f87bc